### PR TITLE
Scheduler

### PR DESCRIPTION
**Pull request by jcant**

### CODE REVIEW

File: lib/common/app_reducer.dart [hunk 1]
```diff
@@ -687,6 +688,24 @@ AppState _changeLessonRequestResponseType(
 }
 
 AppState _changeSchedule(AppState appState, GetScheduleAction action) {
+  List<Lesson> lessons = [];
+
+  action.events?.forEach((event) {
```
Comment (NooAn) [inline lib/common/app_reducer.dart:693]: Maybe this mapper should move to Middleware?

File: lib/resources/id_values.dart [hunk 2]
```diff
@@ -364,6 +371,12 @@ class ForIdValues {
       StringId.NEWS_AD_FORM_SUBMITTED_TITLE:
           'Заявка отправлена! Скоро с тобой свяжется наш менеджер.',
       StringId.LOGOUT_BUTTON: 'Выйти из аккаунта',
+      StringId.SCHEDULE_SCREEN_HEADER: 'Расписание',
+      StringId.TODAY: 'Сегодня',
+      StringId.DAY_AND_MONTH_FORMAT: 'd MMMM',
+      StringId.DAY_MONTH_YEAR_FORMAT: 'd MMMM y г.,',
```
Comment (NooAn) [inline lib/resources/id_values.dart:377]: Are you sure? this is the place for this kind of constant?

Comment (NooAn) [reply lib/resources/id_values.dart:377]: or this date depends on the locale?

File: lib/ui/main_screen.dart [hunk 3]
```diff
@@ -642,55 +662,70 @@ class MainViewModel {
...
+              start: e.start))
+          .toList(),
+      isTrialStudent: store.state.student?.isTrialStudent ?? true,
+      hasNextLesson: store.state.scheduleState.nextLessonDateTime != null,
+      nextLessonDateTime: store.state.scheduleState.nextLessonDateTime,
+      studentName: store.state.student?.name ?? "",
+      isLoading: store.state.isLoading,
+      isEmptyBalance: store.state.mainScreenState.balance == 0,
+      balance: store.state.mainScreenState.balance.toString(),
+      openAddNewWords: () {
```
Comment (NooAn) [inline lib/ui/main_screen.dart:718]: that's good

File: lib/ui/schedule_screen.dart [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+                color: fontColor,
+                borderRadius: BorderRadius.circular(4),
+              ),
+            ),
+          ))
+        : (Container(width: 0, height: 0));
+    Intl.defaultLocale = viewModel.locale;
+    String weekDay = DateFormat.E().format(date);
+    return InkWell(
+      onTap: () => setState(() {
```
Comment (NooAn) [inline lib/ui/schedule_screen.dart:154]: we don't use setState in Redux, please remove it

File: lib/ui/schedule_screen.dart [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+          ]),
+        ),
+      ),
+    );
+  }
+
+  List<Widget> _getEventCards(ScheduleScreenViewModel viewModel) {
+    List<Widget> result = [];
+
+    List<Lesson> selectedDayLessons = viewModel.lessons
```
Comment (NooAn) [inline lib/ui/schedule_screen.dart:203]: maybe move it in viewModel?

File: lib/resources/id_values.dart (hunk 2 above)
Comment (jcant) [reply lib/resources/id_values.dart:377]: At the moment it does. I wasn't sure is it necessary

File: lib/common/app_reducer.dart (hunk 1 above)
Comment (jcant) [reply lib/common/app_reducer.dart:693]: done

File: lib/ui/schedule_screen.dart (hunk 4 above)
Comment (jcant) [reply lib/ui/schedule_screen.dart:154]: done

File: lib/ui/schedule_screen.dart (hunk 5 above)
Comment (jcant) [reply lib/ui/schedule_screen.dart:203]: done

File: lib/resources/id_values.dart (hunk 2 above)
Comment (NooAn) [reply lib/resources/id_values.dart:377]: I don't know too, now we use Spanish locale for date time

**Pull request merged**